edb/pgsql/compiler/astutils.py

```diff
@@ -198,6 +198,81 @@ def extend_binop(
     return result
 
 
+def extend_concat(
+    expr: str | pgast.BaseExpr, *exprs: str | pgast.BaseExpr
+) -> pgast.BaseExpr:
+    return extend_binop(
+        pgast.StringConstant(val=expr) if isinstance(expr, str) else expr,
+        *[
+            pgast.StringConstant(val=e) if isinstance(e, str) else e
+            for e in exprs
+        ],
+        op='||',
+    )
+
+
+def extend_select_op(
+    stmt: Optional[pgast.SelectStmt],
+    *stmts: pgast.SelectStmt,
+    op: str = 'UNION',
+) -> Optional[pgast.SelectStmt]:
+    stmt_list = list(stmts)
+    result: pgast.SelectStmt
+
+    if stmt is None:
+        if len(stmt_list) == 0:
+            return None
+        result = stmt_list.pop(0)
+    else:
+        result = stmt
+
+    for s in stmt_list:
+        if s is not None and s is not result:
+            result = pgast.SelectStmt(larg=result, op=op, rarg=s)
+
+    return result
+
+
+def conditional_string_agg(
```

Comment:
This is obscure enough that I'd probably just stick it in `dml.py` next to the call site